extensions/grpc-common/deployment/src/main/java/io/quarkus/grpc/common/deployment/GrpcCommonProcessor.java

```diff
@@ -75,7 +75,8 @@ NativeImageConfigBuildItem nativeImageConfiguration() {
                 .addRuntimeInitializedClass("io.grpc.netty.Utils")
                 .addRuntimeInitializedClass("io.grpc.netty.NettyServerBuilder")
                 .addRuntimeInitializedClass("io.grpc.netty.NettyChannelBuilder")
-                .addRuntimeInitializedClass("io.grpc.internal.RetriableStream");
+                .addRuntimeInitializedClass("io.grpc.internal.RetriableStream")
+                .addRuntimeReinitializedClass("com.google.protobuf.UnsafeUtil");
```

Comment:
What's the diff between Initialized and Reinitialized class ?

Reply:
Re-initialized re-run the static initialization at runtime.

In this case, if you use runtime initialized, a lot of the things we do at build time would need to be at runtime (because all the users of the UnsafeUtil would need to be changed). With re-initialized, we will recompute the offset at runtime (which is the root of the issue). 

So https://github.com/protocolbuffers/protobuf/blob/main/java/core/src/main/java/com/google/protobuf/UnsafeUtil.java#L21-L59 will be recomputed at runtime.